File: CodingSamples/DotNet/Advanced/Communication/ServerTest/DemoApp2/Services/Shopping/InventoryLookup.cs
using DemoApp.Services.Common;
using Microsoft.Extensions.DependencyInjection;

namespace DemoApp.Services.Shopping;

public class InventoryLookup(IDataAccessLayer storeroom, [FromKeyedServices("wholesale")] IDataAccessLayer warehouse) : ICommunicationLayer
{

    public async Task CommunicateAsync(StreamReader input, StreamWriter output)
    {
        await output.WriteLineAsync("Welcome to MET-DIGITAL supplies.");
        string name = await input.ReadLineAsync();
        var info = storeroom.ReadItemInfo(name) ?? warehouse.ReadItemInfo(name);
        if(info != null)
        {
            await output.WriteLineAsync($"COST is {info.Cost} with STOCK of {info.Stock}");
        }
    }
}

File: CodingSamples/Java/Advanced/Communication/ServerTest/demo2/src/main/java/app/services/shopping/InventoryLookup.java
package app.services.shopping;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.PrintWriter;

import app.services.common.CommunicationLayer;
import jakarta.inject.Inject;
import jakarta.inject.Named;

public class InventoryLookup implements CommunicationLayer {

    @Inject
    private DataAccessLayer storeroom;

    @Inject
    @Named("wholesale")
    private DataAccessLayer warehouse;

    @Override
    public void communicate(BufferedReader input, PrintWriter output) throws IOException {
        output.println("Welcome to MET-DIGITAL supply.");
        String name = input.readLine();
        var info = storeroom.readItemInfo(name);
        if(info == null)
            info = warehouse.readItemInfo(name);
        if(info != null){
            output.printf("COST is %s and STOCK is %s%n", info.cost(), info.stock());
        }
    }
    
    
}
